### ISSUE TITLE
Async module registration may init components multiple times

### ISSUE THREAD

**Issue by nkbt** [packetloop/angular-webpack#2]
Angular has internal cache of initialized components. If I include one module as dependency in several modules (one sync, one async) it gives an exception.
The reason - `<match>` directive was initialized by angular and then by app.register again.

```
Error: [$compile:multidir] Multiple directives [match, match] asking for new/isolated scope on: 

<input name="confirm" type="password"
  ng-model="password.data.user.password_confirmation"
  placeholder="Confirm Password"
  match="password.data.user.password"
  class="ng-pristine ng-untouched ng-valid”>
```

I see 2 possible ways. Either find find the way to access that internal Angular cache, so I will not init same directive more then once. Or always use my own app.register to take full control over caching.

**Comment by nkbt** [comment]
https://github.com/angular/angular.js/pull/4694 I end up using this solution and keep local copy of modified angularjs with patch applied. Works perfectly.

**Comment by nkbt** [comment]
Fixed by #3, added modified angular.js source into the project.
